GirafIntegrationTest/testlib.py

```diff
@@ -56,6 +56,15 @@ def compare_handler(a, b):
     return ordered_handler, compare_handler
 
 
+def is_sequence(obj):
+    try:
+        len(obj)
+        obj[0:0]
+        return True
+    except TypeError:
+        return False
+
+
```

Comment:
This can be done in a much more readable and pythonic manner with

```py
from typing import Sequence, Any

def is_sequence(obj: Any) -> bool:
    return isinstance(obj, Sequence)
```

Also, type hints are nice.

Reply:
> This can be done in a much more readable and pythonic manner with
> 
> ```python
> from typing import Sequence, Any
> 
> def is_sequence(obj: Any) -> bool:
>     return isinstance(obj, Sequence)
> ```
> 
> Also, type hints are nice.

Are you sure that it shouldn't be `collections.Sequence`? Also, I'm pretty sure `isinstance('string', Sequence)` returns true

Reply:
You can use either, but if you use the one from `collections`, use `collections.abc.Sequence`, as the one you mention is deprecated.

And yes, that's because a string is iterable/a sequence. Your original function would also return true on a string.